src/main/java/com/hubspot/guice/transactional/TransactionalDataSource.java

```diff
@@ -92,4 +104,23 @@ public Logger getParentLogger() throws SQLFeatureNotSupportedException {
       return delegate.unwrap(type);
     }
   }
+
+  private void throwIfConnectionInvalid(TransactionalConnection connection) throws SQLException {
+    String dbName = getDbName();
+    if (connection.getDatabaseName() != dbName) {
+      dbName = dbName != null ? dbName : "unknown";
+      throw new SQLException(String.format("During transaction for database %s a query was made to attempt to acquire connection from database %s", connection.getDatabaseName(),
```

Comment:
"Attempt to acquire connection to database %s, during transaction in database %s", dbName, connection.getDatabaseName()

just cleaning up the wording (but it also switches the order of the args to format)